src/App.jsx

```diff
@@ -13,46 +15,122 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const preparedGoods
+  = [...goodsFromServer]
+    .map((good, index) => ({ good, key: index }));
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alpha';
+const SORT_BY_LENGTH = 'length';
+const REVERSE = 'reverse';
+const RESET = 'reset';
+
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(preparedGoods);
+  const [resetButtonHidden, setResetButtonHidden] = useState(false);
+  const [sortField, setSortField] = useState('');
+
+  function getSortBy(products, sortBy) {
+    if (sortBy === SORT_BY_ALPHABET) {
+      // eslint-disable-next-line no-console
+      console.log(products)
+
+      return () => {
+        setResetButtonHidden(true);
```

Comment:
Test failed.

![image](https://github.com/mate-academy/react_list-of-goods-js/assets/91376823/3d6316df-9271-4c05-a76f-42ff8715530c)